ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py

```diff
@@ -0,0 +1,114 @@
+                        "ptp": get(link, "ptp"),
+                        "mac_security": get(link, "mac_security"),
+                        "type": "routed",
+                        "ipv6_enable": get(link, "ipv6_enable"),
+                        "link_tracking_groups": get(link, "link_tracking_groups"),
+                    }
+                )
+
+                # MPLS
+                if self._mpls_lsr is True:
+                    mpls_dict = {"ip": True}
+                    if self._ldp is True:
+                        mpls_dict["ldp"] = {"interface": True, "igp_sync": True}
+
+                    ethernet_interface["mpls"] = mpls_dict
+
+                # IP address
+                if get(link, "ip_address") is not None:
+                    if "unnumbered" in link["ip_address"].lower():
+                        ethernet_interface["ip_address"] = link["ip_address"]
+                    else:
+                        ethernet_interface["ip_address"] = f"{link['ip_address']}/31"
+
+                if self._underlay_routing_protocol == "ospf":
+                    ethernet_interface["ospf_network_point_to_point"] = True
+                    ethernet_interface["ospf_area"] = self._underlay_ospf_area
+
+                if self._underlay_routing_protocol == "isis":
+                    ethernet_interface.update(
+                        {
+                            "isis_enable": self._isis_instance_name,
+                            "isis_metric": default(get(self._hostvars, "isis_default_metric"), 50),
+                            "isis_network_point_to_point": True,
+                            "isis_circuit_type": get(self._hostvars, "isis_default_circuit_type"),
+                        }
+                    )
+
+                if get(link, "underlay_multicast") is True:
+                    ethernet_interface["pim"] = {"ipv4": {"sparse_mode": True}}
+
+            # L2 interface
+            elif get(link, "type") == "underlay_l2":
+                ethernet_interface.update(
+                    {
+                        "type": "switched",
+                    }
+                )
+
+                if (channel_group_id := get(link, "channel_group_id")) is not None:
+                    ethernet_interface["channel_group"] = {
+                        "id": int(channel_group_id),
+                        "mode": "active",
+                    }
+                else:
+                    vlans = default(get(link, "vlans"), [])
+                    ethernet_interface.update(
+                        {
+                            "vlans": list_compress(vlans),
+                            "service_profile": get(self._hostvars, "p2p_uplinks_qos_profile"),
+                            "link_tracking_groups": get(link, "link_tracking_groups"),
+                        }
+                    )
+
+            # Structured Config
+            ethernet_interface.update(default(get(link, "structured_config"), {}))
```

Comment:
I don't think we want to apply the structured_config here. You should set the `struct_cfg` key IIRC. The config will be applied later, so we ensure it comes "last"

Reply:
actually it should not be set here at all for L2interface - this is why storm-contrl is showing up